packages/user/Invite/include/services/user/Invite.hpp

```diff
@@ -13,102 +13,79 @@ namespace UserService
-         uint32_t createInvite(Spki                                  inviteKey,
-                               std::optional<uint32_t>               secondaryId,
-                               std::optional<std::string>            secret,
-                               std::optional<psibase::AccountNumber> app,
-                               std::optional<std::string>            appDomain);
-
-         /// Called by existing Psibase accounts to accept an invite without creating
-         /// a new Psibase account
-         void accept(uint32_t inviteId);
-
-         /// Called by the system account "invited-sys" to accept an invite and
-         /// simultaneously create the new account 'acceptedBy', which is
-         /// authenticated by the provided 'newAccountKey' public key
+         /// - `inviteId` is the id of the invite (could be randomly generated)
+         /// - `inviteKey` is the public key of the invite
+         /// - `numAccounts` is the number of accounts this invite can be used to create
+         /// - `useHooks` is a flag that indicates whether to use hooks to notify the caller when
+         ///    the invite is updated
+         /// - `secret` is an encrypted secret used to redeem the invite
          ///
-         /// Each invite may be used to redeem a maximum of one new account.
-         void acceptCreate(uint32_t               inviteId,
-                           psibase::AccountNumber acceptedBy,
-                           Spki                   newAccountKey);
-
-         /// Called by existing accounts or the system account "invited-sys" to reject
-         /// an invite. Once an invite is rejected, it cannot be accepted or used to
-         /// create a new account
-         void reject(uint32_t inviteId);
-
-         /// Used by the creator of an invite to delete it. Deleted invites are removed
-         /// from the database. An invite can be deleted regardless of whether it has been
-         /// accepted, rejected, or is still pending
+         /// If `useHooks` is true, the caller must be an account with a service deployed on it
+         /// that implements the InviteHooks interface.
+         uint32_t createInvite(uint32_t    inviteId,
+                               Spki        inviteKey,
+                               uint16_t    numAccounts,
+                               bool        useHooks,
+                               std::string secret);
+
+         /// Called by an invite credential (not a user account) to create the specified
+         /// account. The new account is authorized by the specified public key.
+         void createAccount(psibase::AccountNumber account, Spki accountKey);
+
+         /// The sender accepts an invite.
+         /// Calling this action also requires that the sender authorizes the transaction with the
+         /// proof for the credential associated with an invite.
+         void accept();
```

Comment:
I don't think deducing the invite id from the transaction is a good idea.
- It isn't obvious what is being accepted
- It makes some dodgy actions that appear safe on their own exploitable
  - Removing a signature is harmless because it'll just make the transaction fail, right?
  - Adding a signature with an explicit private key is safe, right?
  - Alas, the combination of the above allows someone to force this version of accept to accept the wrong invite.